### PR TITLE
feat: terraform provider for a binding user on mysql

### PR DESCRIPTION
**Pull request by jhvhs**

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
```
Comment (blgm) [inline .github/workflows/golangci-lint.yml:20]: Has a change been made to renovator to update this automatically?

File: Makefile [hunk 2]
```diff
@@ -1,13 +1,35 @@
-default: install
+.DEFAULT_GOAL = help
 
-generate:
-	go generate ./...
+GO-VERSION = 1.19.2
```
Comment (blgm) [inline Makefile:3]: Again, will this be updated automatically?

File: csbmysql/csbmysql_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Terraform Provider CSB MySQL Suite")
+}
+
+var _ = BeforeSuite(func() {
+	By("Building provider")
+	mustRun("go", "build", "..")
+
+	By("Starting MySQL server")
+	mysqlVersion, ok := os.LookupEnv("TEST_MYSQL_VERSION_IMAGE_TAG")
```
Comment (blgm) [inline csbmysql/csbmysql_suite_test.go:42]: If the version of MySQL matters, then GitHub Actions could be used to create a test matrix of different versions.

File: csbmysql/resource_binding_user.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+		CreateContext: resourceBindingUserCreate,
+		ReadContext:   resourceBindingUserRead,
+		UpdateContext: resourceBindingUserUpdate,
+		DeleteContext: resourceBindingUserDelete,
+		Description:   "TODO",
```
Comment (blgm) [inline csbmysql/resource_binding_user.go:43]: What needs to be done here?

File: tools/tools.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+//go:build tools
+// +build tools
+
+package tools
+
+import (
+	_ "github.com/onsi/ginkgo/v2/ginkgo/generators"
+	_ "github.com/onsi/ginkgo/v2/ginkgo/internal"
```
Comment (blgm) [inline tools/tools.go:8]: Why not just import "github.com/onsi/ginkgo/v2/ginkgo"? Importing an "internal" package looks very wrong.

**Review by blgm — CHANGES_REQUESTED** [review]
Generally looks good and is similar to the PostgreSQL provider. From a personal taste perspective, I think it violates the Single Responsibility Principal because the provider has the responsibilities of:
- CRUDing users in MySQL
- Being a Terraform resource

My preference is to split these into different packages, which I did in the MS-SQL provider. It improves testability, and was much faster to develop. But that's a matter of taste and I don't think it's worth re-writing at this stage.

There are two things that I think should be tidied up before merging, and these are:
- Remove the TODO
- Stop importing an "internal" package

File: tools/tools.go (hunk 5 above)
Comment (jhvhs) [reply tools/tools.go:8]: I guess this was some quirk with `go mod tidy` missing some dependencies, so we couldn't run ginkgo with `go run github.com/onsi/ginkgo...`

File: csbmysql/resource_binding_user.go (hunk 4 above)
Comment (jhvhs) [reply csbmysql/resource_binding_user.go:43]: Apparently, that's what we have in the postgresql binding provider :-S

File: Makefile (hunk 2 above)
Comment (jhvhs) [reply Makefile:3]: Yes

File: csbmysql/csbmysql_suite_test.go (hunk 3 above)
Comment (jhvhs) [reply csbmysql/csbmysql_suite_test.go:42]: Done

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (jhvhs) [reply .github/workflows/golangci-lint.yml:20]: Yes

**Pull request merged**